src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+
+    consec_const_steps = 0 # counts how many consecutives steps have been taken
+                           # with same step size
+
+    # Declaration of  four local error (LE) or truncation errors necessary
+    # for variable order calculations, which correspond to infinity norms of:
+    #      LE(n+1)_{k-1}, LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2}
+    # from Hairer pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0, 0.0, 0.0, 0.0
+
+    #-##########################################################################
+    # Variable Step Variable Order (VSVO) Step Integration Loop
+    #-##########################################################################
+    while  tfinal*tdir > t[n]*tdir
+        #--#####################################################################
+        # Attempt a step with current trial stepsize `dt`
+        #--#####################################################################
+        successful_step = false
+        while !successful_step
+            #---################################################################
+            # Make an approriate time step with current trial stepsize `dt`
+            #---################################################################
+            #Make the step
+            t_np1 = t[n]+dt
+            #---################################################################
+            # Calculating explicit approximatiion to y_np1 with current step dt
+            #---################################################################
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g, c, dt, t, t_np1, n, min(n, steporder+2))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar, dy[n], t, t_np1, n, 1:min(max(n-1, 1), steporder+1))
+            #Note that  1<= min(max(n-1,1),steporder+1) <= n-1 and steporder+1
+
+            # Calculate next value, y_np1, with order = k = steporder + 1. See
+            # equation (5.5) in Hairer et al, Vol 1, where we note that here
+            # y_np1 = y_{k}(t_{n+1})
+
+            #y_np1 = copy(y[n])
+            copy!(y_np1, y[n])
+            for j= 1:steporder
+                y_np1 += dt * g[j] * ϕstar[2, j]
+            end
+
+            # Compute current estimate for derivative
+            dy_np1=F(t_np1, y_np1)
+
+            #---################################################################
+            # Correction-Evaluaion (CE) Loop for implicit solver
+            #     See Hairer p. 360 for a general description of this method
+            #---################################################################
+
+            # Set CE_loops to be how many correction-evaulation cycles
+            # desired. As least one loop is necessary, for the sake of the
+            # adaptive step selection
+            CE_loops = 1
+            if n>=steporder+1
```

Comment:
Shouldn't the trapezoidal rule be used on the first step?  But this only runs on the second step, right?